src/printutils.cc

```diff
@@ -20,12 +20,32 @@ bool OpenSCAD::rangeCheck = false;
 
 boost::circular_buffer<std::string> lastmessages(5);
 
+namespace {
+	bool no_throw;
+	bool deferred;
+}
```

Comment:
These are global variables, but AFAIU, the tracking of deferred warnings is per open document. Not sure how to solve that without passing a whole lot more context to the print function.

Reply:
The print stuff already relies on it not switching documents mid event. If it does (which it can do) the output ends up going to the command line console because the interrupting thread changes the output and then clears it. I don't think this PR makes that any worse.

The only way I can think of making PRINT work properly is to pass an object as the first parameter that holds the destination window and these variables. All instances of PRINT will need to be changed and this output window object would need to be passed to all the functions that print. Potentially a huge change to pass the object all the way down.

I think that is the only way to ensure the output goes to the right place when one event can interrupt another when process_events is called or when a thread is used to during F6.